pkg/cloudprovider/cloudapi/azure/azure_compute.go

```diff
@@ -90,11 +105,27 @@ func newComputeServiceConfig(account types.NamespacedName, service azureServiceC
 		vnetAPIClient:          vnetAPIClient,
 		resourceGraphAPIClient: resourceGraphAPIClient,
 		resourcesCache:         &internal.CloudServiceResourcesCache{},
+		vmResourceCache:        make(map[string]*internal.CloudServiceResourcesCache),
 		inventoryStats:         &internal.CloudServiceStats{},
 		credentials:            credentials,
-		computeFilters:         make(map[string][]*string),
-		selectors:              make(map[string]*crdv1alpha1.CloudEntitySelector),
 	}
+
+	config.selectorIndexer = cache.NewIndexer(
+		func(obj interface{}) (string, error) {
+			qs := obj.(*selector)
+			return qs.selectorNamespacedName, nil
+		},
+		cache.Indexers{
+			selectorChanged: func(obj interface{}) ([]string, error) {
+				qs := obj.(*selector)
+				if qs.configChanged {
```

Comment:
dont store changed inside Selector.. its not clean to read it.. create a new slice of selector changed and at each poll, check if slice is empty or not. if empty, do regular polling, if not empty, just poll for that selector only..